lib/ruby_lsp/addon.rb

```diff
@@ -132,6 +170,11 @@ def deactivate; end
     sig { abstract.returns(String) }
     def name; end
 
+    # Add-ons should override the `version` method to return a semantic version string representing the add-on's
+    # version. This is used for compatibility checks
+    sig { abstract.returns(String) }
```

Comment:
Are we anticipating the add-on would be versioned separately from the gem its contained within? Or would we return the gem's own `VERSION` is most cases?

Reply:
What'd happen when the `version` is called on a add-on that doesn't use Sorbet itself? Would it raise an error or return `nil`?